flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/TypesManager.java

```diff
@@ -343,121 +343,121 @@ private void addToMapIfForeignAndNotAdded(Map<String, Symbol> foreignSymbols, Ty
         }
     }
 
-    private void addDependencyTypes(ServiceDeclarationSymbol serviceDeclarationSymbol, Map<String, Object> references) {
+    private void addDependencyTypes(ServiceDeclarationSymbol serviceDeclarationSymbol, Map<String, Object> references, boolean skipParameters) {
         // attributes
-        serviceDeclarationSymbol.fieldDescriptors().forEach((key, field) -> {
-            addDependencyTypes(field.typeDescriptor(), references);
-        });
+        if (!skipParameters) {
+            serviceDeclarationSymbol.fieldDescriptors().forEach((key, field) -> {
+                addDependencyTypes(field.typeDescriptor(), references, skipParameters);
+            });
+        }
 
         // methods
         serviceDeclarationSymbol.methods().forEach((key, method) -> {
-            // params
-            method.typeDescriptor().params().ifPresent(params -> params.forEach(param -> {
-                addDependencyTypes(param.typeDescriptor(), references);
-            }));
+            // params - skip for GraphQL
```

Comment:
Comment - `// params - skip for GraphQL` is not necessary detail for here. The user may pass `true` or `false` to the `addDepndencyTypes`'s argument to control the behaviour. One thing you can do instead of this is - mention GraphQL as an example: `// Skipping Parameters. e.g. GraphQL service class methods`